hles/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DemographicsTransformations.scala

```diff
@@ -56,10 +56,28 @@ object DemographicsTransformations {
   def mapAge(rawRecord: RawRecord, dog: HlesDogDemographics): HlesDogDemographics =
     rawRecord.getOptionalBoolean("dd_dog_birth_year_certain").fold(dog) { ageCertain =>
       if (ageCertain) {
-        val formYear = rawRecord.getRequired("dd_dog_current_year_calc").toInt
+        val dogId = rawRecord.id
+        val formYear = rawRecord.getOptional("dd_dog_current_year_calc") match {
+          case Some(year) => year.toInt
+          case None       =>
+            // we're throwing these errors instead of logging them because we want to completely skip records missing this value. they get logged
+            // by the logic up the chain that catches the error.
+            throw MissingCalcFieldError(
+              s"Record $dogId has less than 1 value for field dd_dog_current_year_calc"
+            )
+        }
+
         val birthYear = rawRecord.getRequired("dd_dog_birth_year").toInt
 
-        val formMonth = rawRecord.getRequired("dd_dog_current_month_calc").toInt
+        val formMonth = rawRecord.getOptional("dd_dog_current_month_calc") match {
+          case Some(month) => month.toInt
+          case None        =>
```

Comment:
we tried to do this with a `map` and an `isEmpty` check, but this way ended up being much cleaner.